### PR TITLE
Feat/keep original domain

### PR DESCRIPTION
**Pull request by tiboun**
## Summary
FEAT

- **BREAKING CHANGE** Add `normalized_domain` variable for schema extraction. `domain` keep the original name.
- keep user changes if set in domain's metadata or table information. Domain-template and other rules that apply to it still have higher precedence.

FIX
- **BREAKING CHANGE** domain template in extract-schema is no more interpolated since we plan to use domains accross multiple environment

### CODE REVIEW

File: src/main/scala/ai/starlake/extract/ExtractSchema.scala [hunk 1]
```diff
@@ -60,39 +59,82 @@ object ExtractSchema extends Extract with LazyLogging {
...
-        case None => table
-        case Some(write) =>
-          val metadata =
-            domainTemplate.flatMap(_.metadata).getOrElse(Metadata()).copy(write = jdbcSchema.write)
-          table.copy(metadata = Some(metadata))
-      }
+      val restoredTable =
+        currentDomain.flatMap(_.tables.find(_.name == table.name)) match {
+          case Some(currentTable) =>
+            table.copy(
```
Comment (hayssams) [inline src/main/scala/ai/starlake/extract/ExtractSchema.scala]: I think that we should put this code in the Schéma class to centralize access to the fields there

File: src/main/scala/ai/starlake/schema/handlers/SchemaHandler.scala [hunk 2]
```diff
@@ -438,18 +448,22 @@ class SchemaHandler(storage: StorageHandler, cliEnv: Map[String, String] = Map.e
             .map { tableRefName =>
               val schemaPath = new Path(folder, tableRefName)
               YamlSerializer.deserializeSchemaRefs(
-                Utils
-                  .parseJinja(storage.read(schemaPath), activeEnv())
-                  .richFormat(activeEnv(), Map.empty),
+                if (raw)
```
Comment (hayssams) [inline src/main/scala/ai/starlake/schema/handlers/SchemaHandler.scala]: Should’nt this be if(!raw)

File: src/main/scala/ai/starlake/schema/model/Metadata.scala [hunk 3]
```diff
@@ -228,6 +247,57 @@ case class Metadata(
     )
   }
 
+  /** Keep metadata that are different only
+    *
+    * @param parent
+    *   : Schema level metadata
+    * @return
+    *   the metadata that differs between parent and this element.
+    */
```
Comment (hayssams) [inline src/main/scala/ai/starlake/schema/model/Metadata.scala:256]: Move this code to Metadata

**Review by hayssams — APPROVED** [review]
Could you please make the changes requested 
Thanks for this PR

File: src/main/scala/ai/starlake/schema/model/Metadata.scala (hunk 3 above)
Comment (tiboun) [reply src/main/scala/ai/starlake/schema/model/Metadata.scala:256]: It's already in Metadata, do you think we should move this to the object Metadata ?

Comment (hayssams) [reply src/main/scala/ai/starlake/schema/model/Metadata.scala:256]: My bad

**Pull request merged**